src/external/settings.hpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @brief Settings structure for class constructors
+ * @details This file contains nested namescpace structs which contain some variables that are 
+ * used in constructor of classes. This ensures the user can create dummy settings object in another file
+ * to pass by reference to a using class. This way the user will not get large object with many variables. 
+ * @file settings.hpp
+ * @author Arsalan Anwari
+ * @date 2018-05-15
+ */
+
+#ifndef SETTINGS_HPP
+#define SETTINGS_HPP
+
+#include "types.hpp"
+
+namespace lucidy{
+
+    namespace settings{
+
+    namespace FDT{
+        
+        /**
+         * @brief Used in Feature detector
+         * 
+         */
+        struct data{
+            double hessianThreshold;
+            int octavesNr = 4;
+            int octaveLayers = 3;
+        };
+
+    }
+
+    namespace FDC{
+        /**
+         * @brief Used in Feature descriptor
+         * 
+         */
+        struct data{
```

Comment:
should be a `class`

Reply:
Why use unnecessary larger data size. If you recommended union comment would be valid. Comment is invalid due to bad advice.

Reply:
Classes are exactly the same as structs, except that their accessibility modifier defaults to private instead of public. Since we are programming in C++ it is standard that we are using classes over structs.